source/gfoidl.Base64/Analyzers/PubternalityAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+// Based on https://github.com/aspnet/Extensions/pull/345
+
+namespace gfoidl.Base64.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class PubternalityAnalyzer : DiagnosticAnalyzer
```

Comment:
Maybe this should be refactored to a separate project, as it can be used more often without the coupling to this project.

Reply:
Here it is: https://github.com/gfoidl/Analyzers